package/src/RNCSliderNativeComponent.web.tsx

```diff
@@ -145,18 +145,36 @@ const RCTSliderWebComponent = React.forwardRef(
       }
     }, [initialValue, updateValue, animatedValue]);
 
-    const onResize = () => {
-      hasBeenResized.current = true;
-    };
     React.useEffect(() => {
+      const invalidateContainerPosition = () => {
+        containerPositionInvalidated.current = true;
+      };
+      const onDocumentScroll = (e: Event) => {
+        if (
+          // Avoid all other checks if already invalidated
+          !containerPositionInvalidated.current &&
+          containerRef.current &&
+          // @ts-ignore
+          e.target.contains(containerRef.current)
+        ) {
+          invalidateContainerPosition();
+        }
+      };
       //@ts-ignore
-      window.addEventListener('resize', onResize);
+      window.addEventListener('resize', invalidateContainerPosition);
+      //@ts-ignore
+      document.addEventListener('scroll', onDocumentScroll, {capture: true});
```

Comment:
If there's no way to do this in a more react-native idiomatic way, my opinion would be to do something like the following at the top of the file:

```ts
// @ts-ignore
const { addEventListener, removeEventListener } = document;

...
const RCTSliderWebComponent = React.forwardRef(
...
);

```

That would reduce the overall number of `@ts-ignore` comments in the code, which aren't ideal.

Perhaps it would be even better to create a unique hook (e.g. `useOnDocumentScroll` that receives a callback or something), that abstracts this functionality.

Reply:
> That would reduce the overall number of @ts-ignore comments in the code, which aren't ideal.

I think the `@ts-ignore` issue (which, again, is pre-existing here) would be best solved by configuring TypeScript to properly typecheck uses of DOM APIs in `.web.tsx` files. This might be solved by adding `/// <reference lib="dom" />` to the top of the file, but I haven't tried it.

Reply:
@jspizziri I've put up #475 to deal with the `@ts-ignore`s comprehensively.